### ISSUE TITLE
Certificate information no longer works

### ISSUE THREAD

**Issue by jhogsett2** [Yelp/yelp-fusion#124]
We have been using the same certificate information in our application for a long time.  Recently it stopped working.  We would like to get these re-activated if possible, because we have a huge conference next week and getting a new build of our mobile application out in time would be impossible.  PLEASE HELP!

**Comment by mittonk** [comment]
@jhogsett2 : 
If you're using Yelp Fusion: What's your Client ID?

If you're using the older Yelp API V2: What's your OAuth Consumer Key?

**Comment by mittonk** [comment]
Also: What actual error message are you getting?  I want to rule out bugs or network problems that might have nothing to do with your credentials.

**Comment by jhogsett2** [comment]
Hi Ken,

                Thanks so much for getting back to me.  We do not get any response back from the web services and it times out.  When we try the same call with a different certificate, it does respond.

Thanks,
Jim Hogsett
Business Analyst
—
ASTM INTERNATIONAL
Helping our world work better
—
100 Barr Harbor Drive, PO Box C700
West Conshohocken, PA 19428-2959, USA
tel +1.610.832.9531  cell +1.484.678.9442  fax +1.610.834.9631
www.astm.org

**Comment by jhogsett2** [comment]
We are using API 2 with the following credentials:

consumerKey: "4z9qt9VWWTe_mfOnQ1oT5A",
consumerSecret: "pQyXcioLb4aMfA91-okw0X1tehw",
accessToken: "g4UIYV2jf9oAFmHucHN7BDV-9veXFnGq",
accessTokenSecret: "z9wy6KiVykTjOki_cTcI-IhiRC0",



Thanks,
Jim Hogsett
Business Analyst
—
ASTM INTERNATIONAL
Helping our world work better
—
100 Barr Harbor Drive, PO Box C700
West Conshohocken, PA 19428-2959, USA
tel +1.610.832.9531  cell +1.484.678.9442  fax +1.610.834.9631
www.astm.org

**Comment by mittonk** [comment]
@jhogsett2 : Great.  (Please don't reveal your `consumerSecret`.)

1. Looks like your credentials from 2012 are still valid and haven't been re-generated recently, good.
2. Don't try to re-generate your credentials, that'll make things worse here.
3. Doesn't look like you've had any successful requests in the last 30 days.  When's the last time your app worked --- last year?  (Sounds like your app is used for a yearly event).

I see a lot of requests using your key, but many are using incorrect OAuth protocols (like supplying `oauth_consumer_secret` and `oauth_token` as parameters directly on the request; those should only go to your local OAuth helper utilities).  I can't tell whether those are coming from your app, or from your desktop debugging attempts.

Here's a silly debugging trick: Could you search for "mittonk_1" in your app?  That'll help me trace the request and see what it looks like from our side.

**Comment by jhogsett2** [comment]
Hey Ken,

                I don’t know when it stopped working. We are a not for profit organization with hundreds of committees.   We have at least one or 2 meetings every month.  December is a big month for us.  The functionality that we use yelp for, may not always be used, so it’s hard to say how long it’s been out.  I doubt more than 2 months though.

My technical resources were trying these credentials from an adhoc app and I’m sure that is some of the requests you were seeing.  From my mobile application, I can’t search for a particular item.   My technical resources could search for mittonk_1 but I can’t and they are in India and off right now.


Thanks,
Jim Hogsett
Business Analyst
—
ASTM INTERNATIONAL
Helping our world work better
—
100 Barr Harbor Drive, PO Box C700
West Conshohocken, PA 19428-2959, USA
tel +1.610.832.9531  cell +1.484.678.9442  fax +1.610.834.9631
www.astm.org

**Comment by jhogsett2** [comment]
Hey Ken,

                I just found a way to send that request through my app.  Go ahead and look for it.

Jim Hogsett
Business Analyst
—
ASTM INTERNATIONAL
Helping our world work better
—
100 Barr Harbor Drive, PO Box C700
West Conshohocken, PA 19428-2959, USA
tel +1.610.832.9531  cell +1.484.678.9442  fax +1.610.834.9631
www.astm.org

**Comment by mittonk** [comment]
1. Hm, don't see that request from you.
2. I downloaded the ASTM app 1.4.1 for iOS and tried some searches --- by choosing a quiet conference and category, I can see my own requests.

For each of them, your app shows "No response received from Yelp.  Please try again later."

For each of them, my tools show we're getting the request (good), and returning a INVALID_PARAMETER error (bad) because of the weird OAuth parameters.

**Comment by jhogsett2** [comment]
We haven’t change that code in years.  Why would it start failing now?

Jim Hogsett
Business Analyst
—
ASTM INTERNATIONAL
Helping our world work better
—
100 Barr Harbor Drive, PO Box C700
West Conshohocken, PA 19428-2959, USA
tel +1.610.832.9531  cell +1.484.678.9442  fax +1.610.834.9631
www.astm.org

**Comment by mittonk** [comment]
Reproduced issue with your credentials from my own test app.  Something's definitely up, digging.

**Comment by jhogsett2** [comment]
Thanks you!!!

Jim Hogsett
Business Analyst
—
ASTM INTERNATIONAL
Helping our world work better
—
100 Barr Harbor Drive, PO Box C700
West Conshohocken, PA 19428-2959, USA
tel +1.610.832.9531  cell +1.484.678.9442  fax +1.610.834.9631
www.astm.org

**Comment by jhogsett2** [comment]
Hey Ken,

                I saw our application is now working.  I really appreciate your help to get this working again.  What was the issue?

Thanks again,
Jim Hogsett
Business Analyst
—
ASTM INTERNATIONAL
Helping our world work better
—
100 Barr Harbor Drive, PO Box C700
West Conshohocken, PA 19428-2959, USA
tel +1.610.832.9531  cell +1.484.678.9442  fax +1.610.834.9631
www.astm.org

**Comment by mittonk** [comment]
@jhogsett2 : 
I found some layers of the onion, and added your key to a whitelist to work around another.  My test app now works, so it's good to hear your real app works as well.

(I found some 3-year-old bogus data related to your key, which probably started surfacing more frequently during some recent refactors --- doesn't look like this was a simple bug in either your code or ours.)

At your leisure, you might want to create a fresh, distinct set of API creds and switch over to those; in this case I'd suggest a whole different Yelp account.  (Don't just hit the "Generate new API 2.0 client/secret" button, that will immediately shut down your old keys and old apps.)

Thanks for reporting this issue, and good luck in December!
